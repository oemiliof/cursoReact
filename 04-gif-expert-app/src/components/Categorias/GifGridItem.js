import React from 'react'

const GifGridItem = ({ id, title, url }) => {

    // console.log(id, title, url)

    return (
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem
