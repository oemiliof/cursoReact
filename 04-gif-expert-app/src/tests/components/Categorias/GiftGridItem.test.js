import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGridItem from "../../../components/Categorias/GifGridItem";

describe("Pruebas en <GiftGtidItem />", () => {
  const title = "Titulo";
  const url = "https://google.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener la imagen igual al url y alt de los props", async () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test("debe de tener animate__fadeIn", async () => {
    const div = wrapper.find("div");

    expect(div.props().className).toContain("animate__fadeIn");
    expect(div.props().className.includes("animate__fadeIn")).toBe(true);
    expect(div.props().className.includes("animate__fadeIn")).not.toBe(false);
    // expect()
  });
});
