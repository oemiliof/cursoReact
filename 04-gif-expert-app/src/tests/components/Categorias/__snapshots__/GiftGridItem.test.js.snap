// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGtidItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Titulo"
    src="https://google.com"
  />
  <p>
    Titulo
  </p>
</div>
`;
